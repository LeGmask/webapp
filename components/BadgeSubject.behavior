When the component is clicked
  A click event is emitted

When the color prop is null
  A "more_horiz" icon is in the component

When the component has variant="dot"
  The span.name is not visible

When the component has [clickable]
  The component is a button
Else
  The component is a span

When the component has :name="null" 
and has not [no-tooltip]
  The component's span.name is not empty (The component has a placeholder)

When [color] is a dark color
  The css`color` is `white`
Else
  The css`color` is `black`

