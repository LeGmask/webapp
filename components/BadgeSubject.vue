<template>
    <div class="BadgeSubject" :style="{backgroundColor: color}">
        <span class="subject-abbr">{{abbreviation.toUpperCase()}}</span>
    </div>
</template>

<script>
export default {
    name: 'BadgeSubject',


    props: {
        color: {
            // hex color format
            validator: function(value) { return value.match(/#(([A-Za-z0-9]{3}){1,2})/) }
        },
        abbreviation: {
            // 3-letters string
            validator: function(value) { return String(value).length === 3 }
        }
    },


    data() {
        return {
            
        }
    },

    computed: {
        
    },


    created() {

    },


    methods: {

    }
}
</script>

<style lang="sass" scoped>
@import '~/assets/defaults'

</style>