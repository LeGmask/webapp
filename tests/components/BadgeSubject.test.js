import { mount } from '@vue/test-utils'
import BadgeSubject from '../../components/BadgeSubject.vue'

describe('BadgeSubject', () => {
  test('is a Vue instance', () => {
    const component = mount(BadgeSubject)
    expect(component.isVueInstance()).toBeTruthy()
  })
  test('is a <button> when clickable', () => {
    const component = mount(BadgeSubject, {
      propsData: {
        clickable: true
      }
    })
    expect(component.is('button')).toBeTruthy()
  })
  test('is a <span> when not clickable', () => {
    const component = mount(BadgeSubject)
    expect(component.is('span')).toBeTruthy()
  })
  test('text is bright when subject color is dark', () => {
    const component = mount(BadgeSubject, {
      propsData: {
        color: '#332299'
      }
    })
    expect(component.attributes().style).toMatch(/color:\s*white/)
  })
  describe("variant 'dot'", () => {
    test('matches snapshot', () => {
      const component = mount(BadgeSubject, { propsData: { variant: 'dot' } })
      expect(component).toMatchSnapshot()
    })
  })
  describe("variant 'badge'", () => {
    test('matches snapshot', () => {
      const component = mount(BadgeSubject, { propsData: { variant: 'badge' } })
      expect(component).toMatchSnapshot()
    })
  })
  test('click event is fired', () => {
    const component = mount(BadgeSubject)
    component.trigger('click')
    expect(component.emitted('click')).toBeTruthy()
  })
})

// test('BadgeSubject emits event when clicked', (t) => {
//   const component = render({})
//   component.trigger('click')
//   t.true(component.emitted('click'))
// })

// test.beforeEach(() => {
//   wrapper = mount(BadgeSubject, {
//     propsData: { BadgeSubject }
//   })
// })

// test('BadgeSubject.vue shapshow', (t) => {
//   t.snapshot({ html: wrapper.html() })
// })

// test('BadgeSubject.vue to display BadgeSubject', (t) => {
//   const $items = wrapper.findAll('li').wrappers

//   $items.forEach(($item, index) => {
//     t.is($item.text(), BadgeSubject[index])
//   })
// })
