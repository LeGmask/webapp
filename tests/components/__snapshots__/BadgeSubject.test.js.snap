// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadgeSubject variant 'badge' matches snapshot 1`] = `<span type="button" class="subject variant-badge"><i class="unknown-icon material-icons-outlined">more_horiz</i><span class="name">Choisir...</span></span>`;

exports[`BadgeSubject variant 'dot' matches snapshot 1`] = `<span type="button" class="subject variant-dot"><i class="unknown-icon material-icons-outlined">more_horiz</i><span class="name">Choisir...</span></span>`;
